### ISSUE TITLE
Jobs are not updated when federated jobs are changed (and status does not show this)

### ISSUE THREAD

**Issue by andersjanmyr** [kubernetes-sigs/kubefed#574]
**What happened**:
When a I update a `FederatedJob`, it tries to update the managed Jobs, but since Jobs are immutable, the update fails silently.

**What you expected to happen**:
I expected one of three things. 
1. The error is propagated back to the `FederatedJob` Status and Events.
2. Or, instead of updating the `Job` it would be replaced (deleted, then created).
3. Or, the `FederatedJob` could itself be immutable, preventing the change altogether.

**How to reproduce it (as minimally and precisely as possible)**:
1. Create a new `FederatedJob` which fails due to missing image or something.
2. Update the `FederatedJob` with a new image.
3. Check that the `Job` is not updated and the error doesn't show up on the `FederatedJob`.

**Anything else we need to know?**:

**Environment**:
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.2", GitCommit:"cff46ab41ff0bb44d8584413b598ad8360ec1def", GitTreeState:"clean", BuildDate:"2019-01-13T23:16:01Z", GoVersion:"go1.11.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.4", GitCommit:"f49fa022dbe63faafd0da106ef7e05a29721d3f1", GitTreeState:"clean", BuildDate:"2018-12-14T06:59:37Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
- Kubernetes version (use `kubectl version`)
$ ./kubefed2 version
kubefed2 version: version.Info{Version:"v0.0.1-alpha.0", GitCommit:"unknown", GitTreeState:"unknown", BuildDate:"unknown", GoVersion:"go1.11.5", Compiler:"gc", Platform:"darwin/amd64"}
```
- Install method: Helm
- Install configuration: cluster-scoped

/kind bug

**Comment by marun** [comment]
Thank you for your detailed report!

#44 describes the same concern related to overrides, but as you point out the same problem can result from an update to immutable fields in the template.  

The error is not silent - it is logged to the controller log - but I agree that it should be surfaced more effectively.   

As to how to resolve the underlying problem of immutability, the propagation controller is not intended to implement type-specific logic. That would likely be the province of a scheduling mechanism, one example of which has been lingering [as a PR](https://github.com/kubernetes-sigs/federation-v2/pull/259) for some time.  I encourage you to chime on on that PR and see if the discussion can be restarted.

**Comment by andersjanmyr** [comment]
I believe solution 1 could be generically implemented. When the error is logged it could also be added as an event on the `Federated<resource>`. It should possibly also change the status.
I can take a look and see how this could be implemented if it's interesting.

**Comment by marun** [comment]
Please go ahead with proposing to add an event to federated resources on propagation failure.  This is something half-heartedly supported today, and could definitely stand improvement.  I'm less convinced of the value of recording errors in status, especially since the same sync controller processes all types and some types are going to commonly report transient propagation failures due to controller interaction (I'm looking at you deployments).

**Comment by marun** [comment]
Proposed solution 1 (indicate the error in events and status) was implemented as part of #740 and the cleanup that proceeded it.  I'm not sure automatic creation/deletion of the job is desirable, but field immutability is likely to be supported now that webhook validation is being implemented.
